src/azure-cli-core/setup.py

```diff
@@ -53,6 +53,7 @@
 
 # TODO These dependencies should be updated to reflect only what this package needs
 DEPENDENCIES = [
+    'adal>=1.0.0',
```

Comment:
Have we figured out why we end up with lower version?
If this issue is only repro'd under dev setup, which i believe is the case, then we should put this in requirements.txt file.
For context, CLI depends on msrestazure to pull in the adal which helps avoid errors caused by diamond dependency
//CC: @lmazuel

Reply:
On a second thought, CLI needs adal with 1.0.2+ , so i don't mind calling out here, but let us listen to Laurent

Reply:
It's fine. Out of curiosity, why do you need 1.0.2? What is interesting inside?

Reply:
Ok, I read the ChangeLog of ADAL. LGTM.